crates/zkvm/hosts/src/sp1.rs

```diff
@@ -1,27 +1,83 @@
 use std::sync::LazyLock;
 
+use cfg_if::cfg_if;
+use strata_primitives::proof::ProofContext;
+#[cfg(feature = "sp1-builder")]
 use strata_sp1_guest_builder::*;
-use zkaleido_sp1_adapter::SP1Host;
+use zkaleido_sp1_host::SP1Host;
 
-use crate::ProofVm;
+pub static ELF_BASE_PATH: LazyLock<String> =
+    LazyLock::new(|| std::env::var("ELF_BASE_PATH").unwrap_or_else(|_| "elfs/sp1".to_string()));
 
-pub static BTC_BLOCKSPACE_HOST: LazyLock<SP1Host> =
-    std::sync::LazyLock::new(|| SP1Host::new_from_pk_bytes(&GUEST_BTC_BLOCKSPACE_PK));
+// BTC_BLOCKSPACE_HOST
+cfg_if! {
+    if #[cfg(feature = "sp1-builder")] {
+        pub static BTC_BLOCKSPACE_HOST: LazyLock<SP1Host> =
+            LazyLock::new(|| SP1Host::new_from_pk_bytes(&GUEST_BTC_BLOCKSPACE_PK));
+    } else {
+        pub static BTC_BLOCKSPACE_HOST: LazyLock<SP1Host> = LazyLock::new(|| {
+            let elf_path = format!("{}/guest-btc-blockspace.elf", &*ELF_BASE_PATH);
+            let elf = std::fs::read(&elf_path)
+                .expect(&format!("Failed to read ELF file from {}", elf_path));
+            SP1Host::init(&elf)
+        });
+    }
+}
 
-pub static EVM_EE_STF_HOST: LazyLock<SP1Host> =
-    std::sync::LazyLock::new(|| SP1Host::new_from_pk_bytes(&GUEST_EVM_EE_STF_PK));
+// EVM_EE_STF_HOST
+cfg_if! {
+    if #[cfg(feature = "sp1-builder")] {
+        pub static EVM_EE_STF_HOST: LazyLock<SP1Host> =
+            LazyLock::new(|| SP1Host::new_from_pk_bytes(&GUEST_EVM_EE_STF_PK));
+    } else {
+        pub static EVM_EE_STF_HOST: LazyLock<SP1Host> = LazyLock::new(|| {
+            let elf_path = format!("{}/guest-evm-ee-stf.elf", &*ELF_BASE_PATH);
+            let elf = std::fs::read(&elf_path)
+                .expect(&format!("Failed to read ELF file from {}", elf_path));
+            SP1Host::init(&elf)
+        });
+    }
+}
 
-pub static CL_STF_HOST: LazyLock<SP1Host> =
-    std::sync::LazyLock::new(|| SP1Host::new_from_pk_bytes(&GUEST_CL_STF_PK));
+// CL_STF_HOST
+cfg_if! {
+    if #[cfg(feature = "sp1-builder")] {
+        pub static CL_STF_HOST: LazyLock<SP1Host> =
+            LazyLock::new(|| SP1Host::new_from_pk_bytes(&GUEST_CL_STF_PK));
+    } else {
+        pub static CL_STF_HOST: LazyLock<SP1Host> = LazyLock::new(|| {
+            let elf_path = format!("{}/guest-cl-stf.elf", &*ELF_BASE_PATH);
+            let elf = std::fs::read(&elf_path)
+                .expect(&format!("Failed to read ELF file from {}", elf_path));
+            SP1Host::init(&elf)
+        });
+    }
+}
 
-pub static CHECKPOINT_HOST: LazyLock<SP1Host> =
-    std::sync::LazyLock::new(|| SP1Host::new_from_pk_bytes(&GUEST_CHECKPOINT_PK));
+// CHECKPOINT_HOST
+cfg_if! {
+    if #[cfg(feature = "sp1-builder")] {
```

Comment:
This seems more verbose than:

```rs
#[cfg(feature = "sp1-builder")]
pub static ...;

#[cfg(not(feature = "sp1-builder"))]
pub static ...;
```

Is there some reason we can't flatten this out?

Same with the others, there's a lot of repetition that *seems* like it could be simplified?